src/SixLabors.Fonts/Tables/General/CMap/Format12SubTable.cs

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) Six Labors and contributors.
+// Licensed under the Apache License, Version 2.0.
+
+using System.Collections.Generic;
+using SixLabors.Fonts.WellKnownIds;
+
+namespace SixLabors.Fonts.Tables.General.CMap
+{
+    internal sealed class Format12SubTable : CMapSubTable
+    {
+        public Format12SubTable(uint language, PlatformIDs platform, ushort encoding, SequentialMapGroup[] groups)
+            : base(platform, encoding, 4)
+        {
+            this.Language = language;
+            this.SequentialMapGroups = groups;
+        }
+
+        public SequentialMapGroup[] SequentialMapGroups { get; }
+
+        public uint Language { get; }
+
+        public override bool TryGetGlyphId(int codePoint, out ushort glyphId)
+        {
+            uint charAsInt = (uint)codePoint;
+
+            for (int i = 0; i < this.SequentialMapGroups.Length; i++)
+            {
+                ref SequentialMapGroup seg = ref this.SequentialMapGroups[i];
+
+                if (charAsInt >= seg.StartCodePoint && charAsInt <= seg.EndCodePoint)
+                {
+                    glyphId = (ushort)((charAsInt - seg.StartCodePoint) + seg.StartGlyphId);
+                    return true;
+                }
+            }
+
+            glyphId = 0;
+            return false;
+        }
+
+        public static IEnumerable<Format12SubTable> Load(IEnumerable<EncodingRecord> encodings, BinaryReader reader)
```

Comment:
I'm curious. Are there performance implications caused by using `IEnumerable<T>`?

Reply:
can't imagine so this is only called during font loading (which should really only be being called once for the life of the running app) and is immediately added to a [List of tables](https://github.com/SixLabors/Fonts/blob/f05fd390be8b276d4cc982aef8ec6274a144d6fc/src/SixLabors.Fonts/Tables/General/CMapTable.cs#L106)